tests/types/test_schema.py

```diff
@@ -1636,6 +1654,60 @@ def test_schema_inference_on_lists(data, data_type):
     assert inferred_schema == expected_schema
 
 
+def test_schema_inference_with_empty_lists():
+    # If (non-nested) empty list is passed, should be inferred as string for backward compatibility
+    # with version 2.7.1: https://github.com/mlflow/mlflow/blob/v2.7.1/mlflow/types/utils.py#L150
+    data = []
+    assert _infer_schema(data) == Schema([ColSpec(DataType.string)])
+
+    # Nested list contains only an empty list is not allowed.
+    data = [[]]
+    with pytest.raises(MlflowException, match=r"Inferred schema contains invalid type `None`"):
+        _infer_schema(data)
```

Comment:
`None` might sound like the given data looks like `data = [[None]]`.

Reply:
Can we use a message like `must include non-empty arrays`?

Reply:
Yeah that sounds better. Might need to consider more general case later but now we only have that failure mode only.